kernelci/api/helper.py

```diff
@@ -512,6 +512,12 @@ def submit_results(self, results, root):
         root_node['result'] = results['node']['result']
         root_node['artifacts'].update(results['node']['artifacts'])
         root_node['data'].update(results['node'].get('data', {}))
+        if root_node['result'] != 'incomplete':
+            data = root_node.get('data', {})
+            if data.get('error_code') == 'node_timeout':
```

Comment:
If the error code is `node_timeout`, then only it will be reset. No other error info will be wiped out.
Moreover, only incomplete nodes will have timeout error. Failed nodes will have different errors.